integration/test_web.py

```diff
@@ -0,0 +1,496 @@
+import sys
```

Comment:
A module docstring would probably be nice.  Maybe this would be a good place to explain the long-term plan for the tests here - ie, something along the lines that they are a rough porting aid and may not necessarily encode behaviors that are critical beyond that point.